lib/baseFactory.js

```diff
@@ -113,6 +112,31 @@ class BaseFactory {
     }
 
     /**
+     * Remove a Model
+     * @method remove
+     * @param  {Mixed}    config               Config object
+     * @return {Promise}
+     */
+    remove(config) {
```

Comment:
@FenrirUnbound Yup this removes a row by id. So this will work for models that don't have any child dependencies. For deleting things like pipeline or job, they need to take advantage of this base `remove` and do extra things. Like how we did for `create` for example: https://github.com/screwdriver-cd/models/blob/master/lib/pipelineFactory.js#L41-L46

I put it in a factory because we have a `create` in here so I think it makes sense to have `remove` in here as well. Also it will be nice to be able to pass in `config` where it could be an `object` contains the key (like in `get`) and not having to get the secret first. Either place is fine with me.